relayer/src/foreign_client.rs

```diff
@@ -406,6 +476,231 @@ impl ForeignClient {
 
         Ok(())
     }
+
+    pub fn update_client_event(
+        &self,
+        consensus_height: Height,
+    ) -> Result<Option<UpdateClient>, ForeignClientError> {
+        let request = QueryClientEventRequest {
+            height: Height::zero(),
+            event_id: IbcEventType::UpdateClient,
+            client_id: self.id.clone(),
+            consensus_height,
+        };
+
+        // Tx query fails if we don't wait a bit here (block is not ready/ indexed?)
+        thread::sleep(Duration::from_millis(100));
```

Comment:
Perhaps add a retry loop here to make the query to `QueryTxRequest` more robust? The `100ms` may work in local tests but not sure how it will fare in actual networks.